assets/app/view/game/scrap_trains.rb

```diff
@@ -22,27 +23,32 @@ def render
             alignItems: 'center',
           },
         }
-
-        h(:div, [
-          h(:h3, 'Trains to Scrap'),
-          h(:div, div_props, scrap_trains(scrappable_trains)),
-        ])
+        if @game.use_compact_scrap_trains_view
+          h(:div, scrap_trains(scrappable_trains))
+        else
+          h(:div,
+            [h(:h3, 'Trains to Scrap'),
+             h(:div, div_props, scrap_trains(scrappable_trains))])
+        end
       end
 
       def scrap_trains(scrappable_trains)
-        step = @game.round.active_step
+        step = @game.round.step_for(@corporation, 'scrap_train')
         scrappable_trains.flat_map do |train|
           scrap = lambda do
             process_action(Engine::Action::ScrapTrain.new(
               @corporation,
               train: train,
             ))
           end
-
-          [h(:div, train.name),
-           h('div.nowrap', train.owner.name),
-           h('div.right', step.scrap_info(train)),
-           h('button.no_margin', { on: { click: scrap } }, step.scrap_button_text(train))]
+          if @game.use_compact_scrap_trains_view
```

Comment:
Calling the method twice doesn't make sense. Just make a separate method that renders the button and have the full render call that method.

Reply:
Done